internal/follower/follower.go

```diff
@@ -114,8 +114,18 @@ func New(ref string, printer *output.Printer, conf *Config) (*Follower, error) {
 		return nil, err
 	}
 
-	// Create temp dir where to put pulled artifacts.
-	tmpDir, err := os.MkdirTemp(conf.TmpDir, "falcoctl-")
+	// Always create temporary directories at the root level to avoid nesting issues.
+	// If TmpDir points to a subdirectory (e.g., /tmp/foo/bar), we use its parent (/tmp/foo)
+	// to prevent creating temporary directories inside artifact directories.
+	baseDir := conf.TmpDir
+	if baseDir != "" {
+		baseDir = filepath.Clean(baseDir)
+		if filepath.Base(baseDir) != "." {
+			baseDir = filepath.Dir(baseDir)
+		}
+	}
+
+	tmpDir, err := os.MkdirTemp(baseDir, "falcoctl-")
```

Comment:
Why do we need this? The temporary dir is given by the user, and is the same for all the followers/artifacts.

Reply:
@TiagoJMartins, could you please provide more info about this change?

Reply:
Oops, this was left over from when I thought this was the problem. Reverted it 👍🏻 

I totally missed this PR comment, sorry about that 🙏🏻